src/wait-for.js

```diff
@@ -34,23 +37,46 @@ function waitFor(
     throw new TypeError('Received `callback` arg must be a function')
   }
 
-  if (interval < 1) interval = 1
-  return new Promise((resolve, reject) => {
-    let lastError
+  return new Promise(async (resolve, reject) => {
+    let lastError, intervalId, observer
+    let finished = false
+
     const overallTimeoutTimer = setTimeout(onTimeout, timeout)
-    const intervalId = setInterval(checkCallback, interval)
 
-    const {MutationObserver} = getWindowFromNode(container)
-    const observer = new MutationObserver(checkCallback)
-    runWithRealTimers(() =>
-      observer.observe(container, mutationObserverOptions),
-    )
-    checkCallback()
```

Comment:
We no longer need to run this with real timers because we only run it when we're not using fake timers.